### PR TITLE
Deleting files

### PR DESCRIPTION
**Pull request by Antonhansel**

### CODE REVIEW

File: lib/crawler.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ Crawler.prototype.processTask = function(folder, cb) {
       var allFilesId = content.body.item_collection.entries.filter(function(item) {
         return item.type !== 'folder';
       }).map(function(item) {
-        return {id: item.id, sha1: item.sha1};
+        return {id: item.id, cursor: self.cursor ? self.cursor : 0};
```
Comment (Neamar) [inline lib/crawler.js]: cursor: self.cursor || 0

File: lib/helpers/getEvents.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        .set('Authorization', 'Bearer ' + serviceData.access_token)
+        .expect(200)
+        .end(cb);
+    },
+    function handleMetada(metadata, cb) {
+      // if this is the first call, just store the new stream position
+      if(serviceData.next_stream_position) {
+        // handling deleted items
+        var deletedFilesId = metadata.body.entries.filter(function(item) {
+          return item.event_type === 'ITEM_TRASH' && item.source.type === 'file';
```
Comment (Neamar) [inline lib/helpers/getEvents.js]: what happen for trashed folders?

File: lib/helpers/getEvents.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          queues.deletion.push(item);
+        });
+      }
+      serviceData.next_stream_position = metadata.body.next_stream_position;
+      cb(null);
+    }
+  ], function(err) {
+    if(err) {
+      log.error(err);
+      logError(err);
```
Comment (Neamar) [inline lib/helpers/getEvents.js]: i believe cb(err) is already piping the error to Opbeat

File: lib/helpers/getEvents.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+'use strict';
```
Comment (Neamar) [inline lib/helpers/getEvents.js]: filename should be get-events

File: lib/helpers/getEvents.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+var log = require('anyfetch-provider').log;
+var logError = require('anyfetch-provider').util.logError;
+var request = require('supertest');
+var async = require('async');
+
+
+module.exports = function upload(serviceData, queues, cb) {
```
Comment (Neamar) [inline lib/helpers/getEvents.js]: not sure about function name

File: lib/update.js [hunk 6]
```diff
@@ -26,13 +27,29 @@ module.exports = function updateAccount(serviceData, cursor, queues, cb) {
...
       // saving the new tokens
-      serviceData = data.body;
-      // init of the crawler
-      var crawler = new Crawler(serviceData, queues, cursor, cb);
-      // push the root folder to the taskQueue to init the crawling
-      crawler.taskQueue.push({id: 0});
+      serviceData.refresh_token = data.body.refresh_token;
+      serviceData.expires_in = data.body.expires_in;
+      serviceData.access_token = data.body.access_token;
+      async.parallel([
```
Comment (Neamar) [inline lib/update.js:33]: no need for a parallel, only one of the two actions should occur

File: lib/update.js [hunk 7]
```diff
@@ -26,13 +27,29 @@ module.exports = function updateAccount(serviceData, cursor, queues, cb) {
...
+            // push the root folder to the taskQueue to init the crawling
+            crawler.taskQueue.push({id: 0});
+          }
+          else {
+            cb();
+          }
+        },
+        function retrieveEvents(cb) {
+          // Retrieve events to get handle deleted/moved files
+          getEvents(serviceData, queues, cb);
```
Comment (Neamar) [inline lib/update.js:48]: maybe you should check for a `next_stream_position`?

File: lib/update.js (hunk 6 above)
Comment (Antonhansel) [inline lib/update.js:33]: Not on the first run

File: lib/update.js (hunk 7 above)
Comment (Antonhansel) [inline lib/update.js:48]: No, on the first run there is no `next_stream_position` , I actually use getEvents to get the first one.

File: lib/helpers/getEvents.js (hunk 2 above)
Comment (Antonhansel) [inline lib/helpers/getEvents.js]: Items within these folders are marked as trashed.

File: test/test-delete.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        }
+      }]
+    });
+
+    done();
+  });
+
+  it('should require a new token, get the root folder and make the first /event call', function(done) {
+    update({refresh_token: 'fake_token', next_stream_position: '1'}, null, fakeQueue, function(err, cursor, serviceData) {
+      cursor.should.not.equal(null);
```
Comment (Neamar) [inline test/test-delete.js:41]: should.not.be.null

Comment (Neamar) [inline test/test-delete.js:41]: (i think)

File: test/test-delete.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      serviceData.access_token.should.equal('fake_access');
+      serviceData.next_stream_position.should.equal('1');
+      fakeQueue.deletion.length.should.equal(1);
```
Comment (Neamar) [inline test/test-delete.js]: fakeQueue.deletion.should.have.lengthOf(1)

File: test/test-crawler.js [hunk 10]
```diff
@@ -50,11 +52,22 @@ describe("Testing the crawler", function() {
         ],
       }
     });
+
+    // mocking the very first event call
+    nock('https://api.box.com:443')
+    .get('/2.0//events?stream_position=now')
```
Comment (Neamar) [inline test/test-crawler.js:58]: still the "//"?

File: lib/update.js (hunk 6 above)
Comment (Neamar) [inline lib/update.js:33]: right

File: test/test-crawler.js (hunk 10 above)
Comment (Antonhansel) [inline test/test-crawler.js:58]: next PR

File: test/test-delete.js (hunk 8 above)
Comment (Antonhansel) [inline test/test-delete.js:41]: Both ways are valid.

File: test/test-crawler.js (hunk 10 above)
Comment (Neamar) [inline test/test-crawler.js:58]: See https://github.com/AnyFetch/box-provider.anyfetch.com/pull/5

**Pull request merged**